packages/react-frontend/src/profile/Profile.test.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { render, fireEvent, screen, act } from '@testing-library/react';
+import '@testing-library/jest-dom/';
+import { BrowserRouter } from 'react-router-dom';
+import Profile from './profile';
+import AuthProvider from '../Auth';
+import { auth } from '../firebase';
+
+// Mock the onAuthStateChanged method
```

Comment:
Good use of mock. I didn't realize this was a thing.

Reply:
I also hadn't heard of Mock. Checked it out and all looks great.